.cloudbuild/library_generation/library_generation.Dockerfile

```diff
@@ -12,11 +12,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# install gapic-generator-java in a separate layer so we don't overload the image
+# with the transferred source code and jars
+FROM gcr.io/cloud-devrel-public-resources/java21 AS ggj-build
+
+WORKDIR /sdk-platform-java
+COPY . .
+# {x-version-update-start:gapic-generator-java:current}
+ENV DOCKER_GAPIC_GENERATOR_VERSION="2.42.1-SNAPSHOT" 
+
+RUN mvn install -DskipTests -Dclirr.skip -Dcheckstyle.skip
+RUN ls "/root/.m2/repository/com/google/api/gapic-generator-java/" 
+RUN cp "/root/.m2/repository/com/google/api/gapic-generator-java/${DOCKER_GAPIC_GENERATOR_VERSION}/gapic-generator-java-${DOCKER_GAPIC_GENERATOR_VERSION}.jar" .
```

Comment:
Only copying the jar allows us for a more precise test. We can use the env vars `DOCKER_GAPIC_GENERATOR_LOCATION` and `DOCKER_GAPIC_GENERATOR_VERSION` to confirm if the preconfigured generator was used

Reply:
Does this jar have all the info we need? Can you test running the image without internet? Sometimes the dependencies of the jar are still downloaded.

Reply:
I ran `docker run --network none` and could get past `generate_library.sh` (i.e. no jars needed), but I found two things we may want to follow up with:

This function from the transferred synthtool fetches data from Maven:
https://github.com/googleapis/sdk-platform-java/blob/0ef66196744e21fc3b4e2553d817c0dea5efe127/library_generation/owlbot/synthtool/languages/java.py#L127

Our script pulls googleapis' proto definitions:
https://github.com/googleapis/sdk-platform-java/blob/0ef66196744e21fc3b4e2553d817c0dea5efe127/library_generation/utils/utilities.py#L175-L182


I thought there was a commit in synthtool that fetches libraries_bom from our config yaml?
https://github.com/googleapis/sdk-platform-java/blob/0ef66196744e21fc3b4e2553d817c0dea5efe127/library_generation/owlbot/synthtool/languages/java.py#L280-L287

For the googleapis download, I guess it's expected behavior

Reply:
I did a temporary bypass (not committed) to see if the generation is successful. It is except for the `mvn fmt:format` part that needs to fetch data when processing a library. The data includes fetching the parent pom of the target library. Nothing about sdk-platform is fetched.

https://paste.googleplex.com/5092731556986880?raw

Reply:
Sounds good! Yes we should definitely but I don't think they should block this PR, since we already have all the jars in the docker image.